functionary/prompt_template/prompt_template_v1.py

```diff
@@ -36,6 +40,17 @@ def get_end_token_from_message(self, message: Dict) -> str:
             else:
                 return self.end_assistant
 
+    def get_start_of_function_call_token(self) -> str:
+        return self.start_function
+
+    def get_stop_token_for_function_parameter(
+        self, stage: Literal["function", "parameter"]
+    ) -> int:
+        if stage == "function":
+            return ":"  # 28747
+        else:
+            return '":'  # 1264
```

Comment:
Oh I think it is only: '"' instead of '":' ?
Because in your check: 
sampled_token== self.get_stop_token_for_function_parameter(stage="parameter")
I assume that: '=:' is 2 tokens?

Reply:
This is for parameter names. The parameters are generated in json format so the models always generated '":' right after it completes a parameter name.